### PR TITLE
[Hold] Allocation autocomplete

### PR DESCRIPTION
**Pull request by nikovacevic**
## What does this PR change?
Adds support for Allocation autocomplete endpoint to the aggregator

## Does this PR rely on any other PRs?
Yes: https://github.com/kubecost/kubecost-cost-model/pull/1928

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Adds allocation autocomplete support

## What risks are associated with merging this PR? What is required to fully test this PR?
Scale of the autocomplete queries could be a problem. Required for testing that is running the harshest queries on a scale data set.

## How was this PR tested?
Testing is in progress

## Have you made an update to documentation? If so, please provide the corresponding PR.
Not yet.

### CODE REVIEW

File: cost-analyzer/templates/cost-analyzer-frontend-config-map-template.yaml [hunk 1]
```diff
@@ -408,6 +408,14 @@ data:
...
             proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
         }
+        location = /model/allocation/autocomplete {
+            proxy_read_timeout          300;
+            proxy_pass http://aggregator/allocation/autocomplete;
+            proxy_redirect off;
+            proxy_set_header Connection "";
+            proxy_set_header  X-Real-IP  $remote_addr;
+            proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
+        }
```
Comment (michaelmdresser) [inline cost-analyzer/templates/cost-analyzer-frontend-config-map-template.yaml:418]: TODO pre-merge: Add caching

Comment (jessegoodier) [reply cost-analyzer/templates/cost-analyzer-frontend-config-map-template.yaml:418]: cache with nginx? need any help? 

https://github.com/kubecost/cost-analyzer-helm-chart/commit/d75698466d187a83c92a42a6a10e91357653073d

note: we should set the size to a reasonable number, 500m in the example above is likely way overkill.

**Comment by chipzoller** [comment]
Added hold and put in draft state. We need a better process for PRs with dependencies on other PRs, in this repo or outside.

**Comment by nikovacevic** [comment]
What do you propose, Chip? In the meantime, I'll close this and reopen when it's ready with a clean git history (not having merges from develop going back into feature branches).

**Comment by chipzoller** [comment]
I would recommend, in the following order of preference:

1. Not sending a PR until its dependency has been merged.
2. Send a PR in draft mode with the title indicating a hold status.
3. Send a PR and assign a label indicating it has a dependency; removing the label when the path is cleared.

**Comment by nikovacevic** [comment]
cc @cliffcolvin @teevans new process proposal, for your consideration :point_up:

**Comment by nikovacevic** [comment]
My two cents:  Firstly, I'm open to changing things with an eye to improvement. Thank you for caring about this process, and for generating recommendations. That said, some PRs are circularly dependent (e.g. OpenCost and Kubecost Cost Model) so recommendation (1) is not possible. With helm PRs, specifically, this is likely not as much of an issue. Circularity not withstanding, sometimes reviewers want to see all the PRs that comprise a feature side-by-side, e.g. this PR. (Michael specifically asked me to open this, to pair with my KCM PR.) So in many cases it's not reasonable to hold the PR entirely until the dependencies are merged. Our practice for the last 4 years has been to open them all simultaneously for a comprehensive review. I see no issue with recommendation (2), at least until the PR is ready for review. I'm skeptical that recommendation (3) will have any impact.

**Pull request closed without merging**